frontend/src/components/views/TUpgradeKubernetesModal/TUpgradeKubernetesModal.vue

```diff
@@ -249,11 +249,20 @@ export default {
           if (
             compareSemVer(v, fromVersion.value) === 0 ||
             compareSemVer(v, fromVersion.value) === 1
-          )
+          ) {
             res.push(v);
+          }
         }
-
-        return res;
+        const currentPatchVer = +fromVersion.value.split(".")[2];
+        for (let v = 0; v < currentPatchVer; v++) {
+          const prevVer = [
+            fromVersion.value.split(".")[0],
+            fromVersion.value.split(".")[1],
+            v,
+          ].join(".");
+          res.push(prevVer);
+        }
+        return res.sort();
```

Comment:
I think it should be done differently:
it should push previous versions during the iteration of the list of the versions.
It's where `compareSemVer` is called.

So for example we can do the following:
```
const parts = fromVersion.value.split(".");
const previous = parseInt(parts[1]) - 1;
const version = `${parts[0]}.${previous}.0`

// then update the filter
          if (
            compareSemVer(v, version) === 0 ||
            compareSemVer(v, version) === 1
          ) {
```